internal/codegen/tpl_stencil.go

```diff
@@ -128,38 +128,39 @@ func (s *TplStencil) GetGlobal(name string) any {
 //
 // This functions write to module hook owned by another module for
 // it to operate on. These are not strongly typed so it's best practice
-// to look at how the owning module uses it for now. Module hooks must always
-// be written to with a list to ensure that they can always be written to multiple
-// times.
+// to look at how the owning module uses it for now.
 //
 //	{{- /* This writes to a module hook */}}
 //	{{- stencil.AddToModuleHook "github.com/myorg/repo" "myModuleHook" (list "myData") }}
-func (s *TplStencil) AddToModuleHook(module, name string, data interface{}) (out string, err error) {
-	k := s.s.sharedState.key(module, name)
-	s.log.With("template", s.t.ImportPath(), "path", k, "data", spew.Sdump(data)).
-		Debug("adding to module hook")
-
-	v := reflect.ValueOf(data)
-	if !v.IsValid() {
-		err := fmt.Errorf("third parameter, data, must be set")
-		return "", err
+func (s *TplStencil) AddToModuleHook(module, name string, data ...any) (out string, err error) {
+	// Attempt to read the destined module's manifest for extra features.
+	var mcfg *configuration.TemplateRepositoryManifest
+	for _, m := range s.s.modules {
+		if m.Name == module {
+			mcfg = m.Manifest
+			break
+		}
 	}
-
-	// we only allow slices or maps to allow multiple templates to
-	// write to the same block
-	if v.Kind() != reflect.Slice {
-		err := fmt.Errorf("unsupported module block data type %q, supported type is slice", v.Kind())
-		return "", err
+	if mcfg != nil {
+		// Check if we have a schema. If we do, use it to validate our module
```

Comment:
is it ok to not have a schema at all then anything goes?

Reply:
i guess it looks like it is from the manifest at bottom, we can see how that goes

Reply:
Correct